libraries/AP_RCProtocol/AP_RCProtocol_Backend.cpp

```diff
@@ -179,6 +179,11 @@ void AP_RCProtocol_Backend::configure_vtx(uint8_t band, uint8_t channel, uint8_t
 void AP_RCProtocol_Backend::log_data(AP_RCProtocol::rcprotocol_t prot, uint32_t timestamp, const uint8_t *data, uint8_t len) const
 {
 #if HAL_LOGGING_ENABLED
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+    if (&rc() == nullptr) { // allow running without RC_Channels if we are doing the examples
```

Comment:
```suggestion
    if (RC_Channels::get_singleton() == nullptr) { // allow running without RC_Channels if we are doing the examples
```

Should probably also remove the header guards or add linux into the mix, as examples can be run as either.

Reply:
I guess ideally we'd have a flag that said it was an example. I didn't want to include the nullptr check for all the builds where we shouldn't need it though. I'll add linux.

Reply:
Done.